ui/StatusQ/src/StatusQ/Layout/StatusAppNavBar.qml

```diff
@@ -1,195 +1,129 @@
 import QtQuick 2.13
-import QtQuick.Layouts 1.13
-import QtQuick.Controls 2.13
-import QtQml.Models 2.13
-import Qt.labs.qmlmodels 1.0
+import QtQuick.Layouts 1.14
 
 import StatusQ.Core 0.1
 import StatusQ.Core.Theme 0.1
 import StatusQ.Controls 0.1
 import StatusQ.Popups 0.1
 
-Rectangle {
-    id: statusAppNavBar
+Item {
+    id: root
 
-    width: 78
-    implicitHeight: 600
-    color: Theme.palette.statusAppNavBar.backgroundColor
+    property alias chatItemsModel: chatItemsListView.model
+    property alias chatItemDelegate: chatItemsListView.delegate
 
-    property var sectionModel: []
-    property string communityTypeRole: ""
-    property int communityTypeValue: -1
-    property int navBarButtonSpacing: 12
+    property alias communityItemsModel: communityItemsListView.model
+    property alias communityItemDelegate: communityItemsListView.delegate
 
-    property StatusNavBarTabButton navBarCameraButton 
-    property StatusNavBarTabButton navBarProfileButton
-    property Component regularNavBarButton
-    property Component communityNavBarButton
+    property alias regularItemsModel: regularItemsListView.model
+    property alias regularItemDelegate: regularItemsListView.delegate
 
-    property var filterRegularItem: function(item) { return true; }
-    property var filterCommunityItem: function(item) { return true; }
+    property real delegateHeight
 
-    signal aboutToUpdateFilteredRegularModel()
-    signal aboutToUpdateFilteredCommunityModel()
+    property alias cameraComponent: cameraItemLoader.sourceComponent
+    property alias profileComponent: profileItemLoader.sourceComponent
 
-    onNavBarProfileButtonChanged: {
-        if (!!navBarProfileButton) {
-            navBarProfileButton.parent = navBarProfileButtonSlot
-        }
+    implicitWidth: 78
+    implicitHeight: layout.implicitHeight
+
+    Rectangle {
```

Comment:
If we need a rectangle, why not keep it as the root?

Reply:
Good point :+1: